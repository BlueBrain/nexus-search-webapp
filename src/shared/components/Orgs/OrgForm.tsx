import * as React from 'react';
import { Form, Input, Button, Spin, Modal } from 'antd';
import { WrappedFormUtils } from 'antd/lib/form/Form';

export interface OrgFormProps {
  form: WrappedFormUtils;
  org?: {
    label: string;
    description?: string;
  };
  busy?: boolean;
  onSubmit?(org: OrgFormProps['org']): any;
  onDeprecate?(): any;
  mode?: 'create' | 'edit';
}

const OrgForm: React.FunctionComponent<OrgFormProps> = ({
  form,
  org,
  mode,
  busy = false,
  onSubmit = () => {},
  onDeprecate = () => {},
}) => {
  const { getFieldDecorator, getFieldValue } = form;
  const formItemLayout = {
    labelCol: {
      xs: { span: 24 },
      sm: { span: 5 },
    },
    wrapperCol: {
      xs: { span: 24 },
      sm: { span: 19 },
    },
  };
  const formItemLayoutWithOutLabel = {
    wrapperCol: {
      xs: { span: 24, offset: 0 },
      sm: { span: 19, offset: 5 },
    },
  };

  const confirmDeprecate = () => {
    Modal.confirm({
      title: 'Deprecate Organization',
      content: 'Are you sure?',
      onOk: onDeprecate,
    });
  };

  const handleSubmit = (e: React.SyntheticEvent) => {
    e.preventDefault();
    form.validateFields((err, values) => {
      if (!err) {
        onSubmit(values);
      }
    });
  };

  return (
    <Spin spinning={busy}>
      <Form onSubmit={handleSubmit}>
        <Form.Item label="Label" {...formItemLayout}>
          {getFieldDecorator('label', {
            initialValue: org ? org.label : '',
            rules: [
              {
                required: true,
                whitespace: true,
                pattern: /^\S+$/g,
                message: 'Label must be a phrase without spaces',
              },
            ],
          })(<Input placeholder="Label" disabled={mode === 'edit'} />)}
        </Form.Item>
        <Form.Item label="Description" {...formItemLayout}>
          {getFieldDecorator('description', {
            initialValue: org ? org.description : '',
            rules: [{ required: false }],
          })(<Input placeholder="Description" />)}
        </Form.Item>
        <Form.Item {...formItemLayoutWithOutLabel}>
          <Button type="primary" htmlType="submit">
            {mode === 'edit' ? 'Save' : 'Create'}
          </Button>
          {mode === 'edit' && (
            <Button
              type="danger"
              onClick={confirmDeprecate}
              style={{ float: 'right' }}
            >
              Deprecate
            </Button>
          )}
        </Form.Item>
      </Form>
    </Spin>
  );
};

export default Form.create()(OrgForm);
