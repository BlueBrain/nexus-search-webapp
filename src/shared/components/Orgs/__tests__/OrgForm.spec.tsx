import * as React from 'react';
import { mount } from 'enzyme';
import OrgForm from '../OrgForm';

const formComponent = <OrgForm />;

const wrapper = mount(formComponent);
describe('Organisation Form component', () => {
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
