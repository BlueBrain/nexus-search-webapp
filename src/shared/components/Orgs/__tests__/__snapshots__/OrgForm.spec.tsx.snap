// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisation Form component should render correctly 1`] = `
<Form(OrgForm)>
  <OrgForm
    form={
      Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      }
    }
  >
    <Spin
      size="default"
      spinning={false}
      wrapperClassName=""
    >
      <div
        className="ant-spin-nested-loading"
      >
        <div
          className="ant-spin-container"
          key="container"
        >
          <Form
            colon={true}
            hideRequiredMark={false}
            layout="horizontal"
            onSubmit={[Function]}
          >
            <form
              className="ant-form ant-form-horizontal"
              onSubmit={[Function]}
            >
              <FormItem
                hasFeedback={false}
                label="Label"
                labelCol={
                  Object {
                    "sm": Object {
                      "span": 5,
                    },
                    "xs": Object {
                      "span": 24,
                    },
                  }
                }
                wrapperCol={
                  Object {
                    "sm": Object {
                      "span": 19,
                    },
                    "xs": Object {
                      "span": 24,
                    },
                  }
                }
              >
                <Row
                  className="ant-form-item"
                  gutter={0}
                >
                  <div
                    className="ant-row ant-form-item"
                  >
                    <Col
                      className="ant-form-item-label"
                      sm={
                        Object {
                          "span": 5,
                        }
                      }
                      xs={
                        Object {
                          "span": 24,
                        }
                      }
                    >
                      <div
                        className="ant-form-item-label ant-col-xs-24 ant-col-sm-5"
                      >
                        <label
                          className="ant-form-item-required"
                          htmlFor="label"
                          onClick={[Function]}
                          title="Label"
                        >
                          Label
                        </label>
                      </div>
                    </Col>
                    <Col
                      className="ant-form-item-control-wrapper"
                      sm={
                        Object {
                          "span": 19,
                        }
                      }
                      xs={
                        Object {
                          "span": 24,
                        }
                      }
                    >
                      <div
                        className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-19"
                      >
                        <div
                          className="ant-form-item-control"
                        >
                          <span
                            className="ant-form-item-children"
                          >
                            <Input
                              data-__field={
                                Object {
                                  "name": "label",
                                }
                              }
                              data-__meta={
                                Object {
                                  "initialValue": "",
                                  "name": "label",
                                  "originalProps": Object {
                                    "disabled": false,
                                    "placeholder": "Label",
                                    "type": "text",
                                  },
                                  "ref": null,
                                  "rules": Array [
                                    Object {
                                      "message": "Label must be a phrase without spaces",
                                      "pattern": /\\^\\\\S\\+\\$/g,
                                      "required": true,
                                      "whitespace": true,
                                    },
                                  ],
                                  "trigger": "onChange",
                                  "validate": Array [
                                    Object {
                                      "rules": Array [
                                        Object {
                                          "message": "Label must be a phrase without spaces",
                                          "pattern": /\\^\\\\S\\+\\$/g,
                                          "required": true,
                                          "whitespace": true,
                                        },
                                      ],
                                      "trigger": Array [
                                        "onChange",
                                      ],
                                    },
                                  ],
                                  "valuePropName": "value",
                                }
                              }
                              disabled={false}
                              id="label"
                              onChange={[Function]}
                              placeholder="Label"
                              type="text"
                              value=""
                            >
                              <input
                                className="ant-input"
                                data-__field={
                                  Object {
                                    "name": "label",
                                  }
                                }
                                data-__meta={
                                  Object {
                                    "initialValue": "",
                                    "name": "label",
                                    "originalProps": Object {
                                      "disabled": false,
                                      "placeholder": "Label",
                                      "type": "text",
                                    },
                                    "ref": null,
                                    "rules": Array [
                                      Object {
                                        "message": "Label must be a phrase without spaces",
                                        "pattern": /\\^\\\\S\\+\\$/g,
                                        "required": true,
                                        "whitespace": true,
                                      },
                                    ],
                                    "trigger": "onChange",
                                    "validate": Array [
                                      Object {
                                        "rules": Array [
                                          Object {
                                            "message": "Label must be a phrase without spaces",
                                            "pattern": /\\^\\\\S\\+\\$/g,
                                            "required": true,
                                            "whitespace": true,
                                          },
                                        ],
                                        "trigger": Array [
                                          "onChange",
                                        ],
                                      },
                                    ],
                                    "valuePropName": "value",
                                  }
                                }
                                disabled={false}
                                id="label"
                                onChange={[Function]}
                                onKeyDown={[Function]}
                                placeholder="Label"
                                type="text"
                                value=""
                              />
                            </Input>
                          </span>
                          <Animate
                            animation={Object {}}
                            component=""
                            componentProps={Object {}}
                            key="help"
                            onAppear={[Function]}
                            onEnd={[Function]}
                            onEnter={[Function]}
                            onLeave={[Function]}
                            transitionAppear={true}
                            transitionEnter={true}
                            transitionLeave={true}
                            transitionName="show-help"
                          />
                        </div>
                      </div>
                    </Col>
                  </div>
                </Row>
              </FormItem>
              <FormItem
                hasFeedback={false}
                label="Description"
                labelCol={
                  Object {
                    "sm": Object {
                      "span": 5,
                    },
                    "xs": Object {
                      "span": 24,
                    },
                  }
                }
                wrapperCol={
                  Object {
                    "sm": Object {
                      "span": 19,
                    },
                    "xs": Object {
                      "span": 24,
                    },
                  }
                }
              >
                <Row
                  className="ant-form-item"
                  gutter={0}
                >
                  <div
                    className="ant-row ant-form-item"
                  >
                    <Col
                      className="ant-form-item-label"
                      sm={
                        Object {
                          "span": 5,
                        }
                      }
                      xs={
                        Object {
                          "span": 24,
                        }
                      }
                    >
                      <div
                        className="ant-form-item-label ant-col-xs-24 ant-col-sm-5"
                      >
                        <label
                          className=""
                          htmlFor="description"
                          onClick={[Function]}
                          title="Description"
                        >
                          Description
                        </label>
                      </div>
                    </Col>
                    <Col
                      className="ant-form-item-control-wrapper"
                      sm={
                        Object {
                          "span": 19,
                        }
                      }
                      xs={
                        Object {
                          "span": 24,
                        }
                      }
                    >
                      <div
                        className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-sm-19"
                      >
                        <div
                          className="ant-form-item-control"
                        >
                          <span
                            className="ant-form-item-children"
                          >
                            <Input
                              data-__field={
                                Object {
                                  "name": "description",
                                }
                              }
                              data-__meta={
                                Object {
                                  "initialValue": "",
                                  "name": "description",
                                  "originalProps": Object {
                                    "disabled": false,
                                    "placeholder": "Description",
                                    "type": "text",
                                  },
                                  "ref": null,
                                  "rules": Array [
                                    Object {
                                      "required": false,
                                    },
                                  ],
                                  "trigger": "onChange",
                                  "validate": Array [
                                    Object {
                                      "rules": Array [
                                        Object {
                                          "required": false,
                                        },
                                      ],
                                      "trigger": Array [
                                        "onChange",
                                      ],
                                    },
                                  ],
                                  "valuePropName": "value",
                                }
                              }
                              disabled={false}
                              id="description"
                              onChange={[Function]}
                              placeholder="Description"
                              type="text"
                              value=""
                            >
                              <input
                                className="ant-input"
                                data-__field={
                                  Object {
                                    "name": "description",
                                  }
                                }
                                data-__meta={
                                  Object {
                                    "initialValue": "",
                                    "name": "description",
                                    "originalProps": Object {
                                      "disabled": false,
                                      "placeholder": "Description",
                                      "type": "text",
                                    },
                                    "ref": null,
                                    "rules": Array [
                                      Object {
                                        "required": false,
                                      },
                                    ],
                                    "trigger": "onChange",
                                    "validate": Array [
                                      Object {
                                        "rules": Array [
                                          Object {
                                            "required": false,
                                          },
                                        ],
                                        "trigger": Array [
                                          "onChange",
                                        ],
                                      },
                                    ],
                                    "valuePropName": "value",
                                  }
                                }
                                disabled={false}
                                id="description"
                                onChange={[Function]}
                                onKeyDown={[Function]}
                                placeholder="Description"
                                type="text"
                                value=""
                              />
                            </Input>
                          </span>
                          <Animate
                            animation={Object {}}
                            component=""
                            componentProps={Object {}}
                            key="help"
                            onAppear={[Function]}
                            onEnd={[Function]}
                            onEnter={[Function]}
                            onLeave={[Function]}
                            transitionAppear={true}
                            transitionEnter={true}
                            transitionLeave={true}
                            transitionName="show-help"
                          />
                        </div>
                      </div>
                    </Col>
                  </div>
                </Row>
              </FormItem>
              <FormItem
                hasFeedback={false}
                wrapperCol={
                  Object {
                    "sm": Object {
                      "offset": 5,
                      "span": 19,
                    },
                    "xs": Object {
                      "offset": 0,
                      "span": 24,
                    },
                  }
                }
              >
                <Row
                  className="ant-form-item"
                  gutter={0}
                >
                  <div
                    className="ant-row ant-form-item"
                  >
                    <Col
                      className="ant-form-item-control-wrapper"
                      sm={
                        Object {
                          "offset": 5,
                          "span": 19,
                        }
                      }
                      xs={
                        Object {
                          "offset": 0,
                          "span": 24,
                        }
                      }
                    >
                      <div
                        className="ant-form-item-control-wrapper ant-col-xs-24 ant-col-xs-offset-0 ant-col-sm-19 ant-col-sm-offset-5"
                      >
                        <div
                          className="ant-form-item-control"
                        >
                          <span
                            className="ant-form-item-children"
                          >
                            <Button
                              block={false}
                              ghost={false}
                              htmlType="submit"
                              loading={false}
                              type="primary"
                            >
                              <Wave>
                                <button
                                  className="ant-btn ant-btn-primary"
                                  onClick={[Function]}
                                  type="submit"
                                >
                                  <span
                                    key=".0"
                                  >
                                    Create
                                  </span>
                                </button>
                              </Wave>
                            </Button>
                          </span>
                          <Animate
                            animation={Object {}}
                            component=""
                            componentProps={Object {}}
                            key="help"
                            onAppear={[Function]}
                            onEnd={[Function]}
                            onEnter={[Function]}
                            onLeave={[Function]}
                            transitionAppear={true}
                            transitionEnter={true}
                            transitionLeave={true}
                            transitionName="show-help"
                          />
                        </div>
                      </div>
                    </Col>
                  </div>
                </Row>
              </FormItem>
            </form>
          </Form>
        </div>
      </div>
    </Spin>
  </OrgForm>
</Form(OrgForm)>
`;
